src/main/webapp/app/atlas/manage/learning-paths-table/learning-paths-table.component.spec.ts

```diff
@@ -130,6 +132,29 @@ describe('LearningPathsTableComponent', () => {
         expect(alertServiceErrorSpy).toHaveBeenCalledOnce();
     });
 
+    it('should load and set average progress for the course', async () => {
+        const mockAverageProgress = { averageProgress: 42, courseId: 1, totalStudents: 5 };
+        jest.spyOn(learningPathApiService, 'getAverageProgressForCourse').mockResolvedValue(mockAverageProgress);
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(learningPathApiService.getAverageProgressForCourse).toHaveBeenCalledWith(courseId);
```

Comment:
Just a nitpick:
you might want to consider adding toHaveBeenCalledOnce/Times()  here as well